.github/workflows/main.yml

```diff
@@ -4,23 +4,30 @@ on: [push, pull_request]
 jobs:
   build:
     name: Run benchmarks
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [windows-latest, macos-latest, ubuntu-latest]
     steps:
-      - uses: actions/checkout@v1
-      - uses: actions/setup-python@v1
-      - name: run benchmark
-        run: |
-             pip install -r requirements.txt
-             asv machine --yes
-             dvc repro run_benchmarks.dvc
-             asv publish
-      - name: upload benchmark results
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+            python-version: 3.7
+      - name: install requirements
+        run: pip install -r requirements.txt
+      - name: setup asv machine
+        run: asv machine --yes
+      - name: run benchmarks
+        run: dvc repro run_benchmarks.dvc
+      - name: convert results to static website
+        run: asv publish
+      - name: upload benchmarks html
```

Comment:
Btw, why are we uploading html and not the results themselves? Those could be then visualized locally. I suppose `results` also includes profiler report, right?

Reply:
When running with `--profile` result json seems to include something like dump of cprofile. Need to verify that.